### PR TITLE
chore: google-cloud-compute-v1 quickstart samples

### PR DESCRIPTION
**Pull request by yuryu**
This PR add a quick start code snippets to google-cloud-compute-v1. The samples are based on the [python](https://github.com/googleapis/python-compute/blob/master/samples/snippets/quickstart.py) version of the same tutorial.

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/google-cloud-ruby/issues) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea.
- [x] Follow the instructions in [CONTRIBUTING](CONTRIBUTING.md). Most importantly, ensure the tests and linter pass by running `bundle exec rake ci` in the gem subdirectory.
- [x] Update code documentation if necessary.

part of #13119

### CODE REVIEW

**Comment by yuryu** [comment]
I ported the same quickstart samples from [Python](https://github.com/googleapis/python-compute/tree/master/samples/snippets), but the delete sample fails with a timeout error.

```

  1) Failure:
ComputeQuickstartTest#test_quickstart [/Users/yuryu/src/google-cloud-ruby/google-cloud-compute-v1/samples/acceptance/quickstart_test.rb:37]:
In stdout.
Expected /Instance i-9154528c6f62 deleted./ to match "Deleting i-9154528c6f62 from us-central1-c...\nNet::ReadTimeout with #<TCPSocket:(closed)>\n\n".

1 runs, 8 assertions, 1 failures, 0 errors, 0 skips
```

I'm not sure if we can change the timeout, but the [doc](https://googleapis.dev/ruby/google-cloud-compute-v1/latest/Google/Cloud/Compute/V1/ZoneOperations/Rest/Client.html#wait-instance_method) suggests we call `#wait` repeatedly until  `operation#status` returns `DONE`. But it also feels like `#wait` shouldn't raise `Net::ReadTimeout` in the first place. What do you think the best option here?

File: google-cloud-compute-v1/samples/quickstart.rb
Comment (viacheslav-rostovtsev) [inline google-cloud-compute-v1/samples/quickstart.rb:152]: Suggestion: name it a zone_operations_client

File: google-cloud-compute-v1/samples/quickstart.rb
Comment (viacheslav-rostovtsev) [inline google-cloud-compute-v1/samples/quickstart.rb:157]: Potential for "infinite" wait without any output at the server's mercy.
Suggestion: loop a fixed amount of retries (e.g. 3) and output something after each try if operation is not complete yet.
Also suggestion: you have this `wait_for_operation` method below, why not upgrade that with a retry logic.

File: google-cloud-compute-v1/samples/quickstart.rb
Comment (viacheslav-rostovtsev) [inline google-cloud-compute-v1/samples/quickstart.rb:159]: In case of error we should probably throw or at least not output `Instance deleted` below, it might not be.

**Review by viacheslav-rostovtsev — COMMENTED** [review]
Looking good, couple suggestions

File: google-cloud-compute-v1/samples/quickstart.rb
Comment (dazuma) [inline google-cloud-compute-v1/samples/quickstart.rb:55]: I know the Python samples use these proto classes and construct proto objects when creating requests. I think the main reason they do so is for IDE autocompletion. However, I think we should use nested hashes to represent requests in the Ruby samples. It's much easier to read, more consistent with our other samples, and feels more Ruby-like.

File: google-cloud-compute-v1/samples/quickstart.rb
Comment (yuryu) [reply google-cloud-compute-v1/samples/quickstart.rb:152]: A bit long? How about operations_client?

File: google-cloud-compute-v1/samples/quickstart.rb
Comment (yuryu) [reply google-cloud-compute-v1/samples/quickstart.rb:157]: Oh wait, I was looking at something else. Yeah that makes sense. I need to figure out how they fit in the tutorial.

File: google-cloud-compute-v1/samples/acceptance/quickstart_test.rb
Comment (viacheslav-rostovtsev) [inline google-cloud-compute-v1/samples/acceptance/quickstart_test.rb:30]: I don't know if you want to wait_until_done here

**Review by viacheslav-rostovtsev — COMMENTED** [review]
Looks good. 
The client is going to be updated soon with the pagination helpers, idk if you want to use them in the examples for `list`/`list_all`

File: google-cloud-compute-v1/samples/acceptance/quickstart_test.rb
Comment (yuryu) [reply google-cloud-compute-v1/samples/acceptance/quickstart_test.rb:30]: Good point, I think you actually don't need to wait here because it's just cleaning up.

File: google-cloud-compute-v1/samples/quickstart.rb
Comment (Xmasreturns) [inline google-cloud-compute-v1/samples/quickstart.rb:54]: In the create_instance sample, other language variations also mention that instance creation requires at least one persistent disk and one network interface. Would we want to duplicate that info in Ruby as well?

File: google-cloud-compute-v1/samples/quickstart.rb
Comment (Xmasreturns) [inline google-cloud-compute-v1/samples/quickstart.rb:36]:
```suggestion
# @param [String] zone name of the zone you want to use. For example: "us-west3-b"
```

File: google-cloud-compute-v1/samples/quickstart.rb
Comment (Xmasreturns) [inline google-cloud-compute-v1/samples/quickstart.rb:94]:
```suggestion
# @param [String] zone name of the zone you want to use. For example: "us-west3-b"
```

File: google-cloud-compute-v1/samples/quickstart.rb
Comment (Xmasreturns) [inline google-cloud-compute-v1/samples/quickstart.rb:137]:
```suggestion
# @param [String] zone name of the zone you want to use. For example: "us-west3-b"
```

File: google-cloud-compute-v1/samples/quickstart.rb
Comment (yuryu) [reply google-cloud-compute-v1/samples/quickstart.rb:59]: Added!

File: google-cloud-compute-v1/samples/README.md
Comment (rsamborski) [inline google-cloud-compute-v1/samples/README.md:9]:
```suggestion
These samples show how to use the [Ruby API Client library](https://github.com/googleapis/google-cloud-ruby/tree/master/google-cloud-compute-v1) for the [Google Cloud Compute Engine API](https://cloud.google.com/compute/docs/reference/rest/v1)
```

**Review by rsamborski — COMMENTED** [review]
LGTM, just a small suggestion added.
@Xmasreturns please feel free to review the comments in regards to the language used and to align them with other languages.

File: google-cloud-compute-v1/samples/README.md
Comment (Xmasreturns) [reply google-cloud-compute-v1/samples/README.md:9]: Instead of `[Ruby API Client library]`, recommend using `[Ruby Cloud Client Library]` to better distinguish from the client library in the googleapis/google-api-ruby-client repo

**Comment by yuryu** [comment]
I need to make changes to support PagedEnumerable.

**Comment by rsamborski** [comment]
LGTM after latest updates

File: google-cloud-compute-v1/samples/acceptance/default_values_test.rb
Comment (viacheslav-rostovtsev) [inline google-cloud-compute-v1/samples/acceptance/default_values_test.rb:39]: So even wait_until_done is not enough?

**Review by viacheslav-rostovtsev — APPROVED** [review]

File: google-cloud-compute-v1/samples/acceptance/default_values_test.rb
Comment (yuryu) [reply google-cloud-compute-v1/samples/acceptance/default_values_test.rb:39]: I followed the python example and didn't test it by myself. I can remove it and see if it works.

Comment (yuryu) [reply google-cloud-compute-v1/samples/acceptance/default_values_test.rb:39]: I tried running it without the sleeps locally and it seems like working fine.

**Comment by yuryu** [comment]
Can someone merge this? I don't have write access.

**Pull request merged**
